### PR TITLE
[jaxrs] Fix response annotation generation (fixes #2588)

### LINKED ISSUE #2588: JAX-RS ApiResponse 'response' and 'responseContainer' annotations get duplicated
I used the Swagger Editor to create:

```
swagger: '2.0'
info:
  version: "0.0.1"
  title: Response annotation test
paths:
 /person:
    get:
      summary: Get all person api
      responses:
        200:
          description: Get all person
          schema:
            type: array
            items:          
              $ref: '#/definitions/Person'
        500:
          description: Unexpected error(s)
          schema:
            $ref: '#/definitions/ApiError'
definitions:
  Person:
    type: object
    properties:
      name:
        type: string
  ApiError:
    type: object
    properties:
      errorMsg:
        type: string
```

Then I generated JAX-RS Server code from the editor.  Here is the APIResponses section in the PersonApi class:

```
   @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "Get all person", response = Person.class, responseContainer = "List"),
        @io.swagger.annotations.ApiResponse(code = 500, message = "Unexpected error(s)", response = Person.class, responseContainer = "List") })
```

The 'response' and 'responseContainer' annotations from the 200 response block got used in the 500 ApiResponse.

### PR DESCRIPTION
**Pull request by amcginn**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

* Added logic in the JaxRS Server generator to translate the CodegenResponse return type and container type
* Updated the JavaJaxRS api.mustache file to correctly output the response type and container type (if applicable)

### CODE REVIEW

File: samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/FakeApi.java [hunk 1]
```diff
@@ -120,13 +120,13 @@ public Response testClientModel(@ApiParam(value = "client model" ,required=true)
     
     @Consumes({ "application/xml; charset=utf-8", "application/json; charset=utf-8" })
     @Produces({ "application/xml; charset=utf-8", "application/json; charset=utf-8" })
-    @io.swagger.annotations.ApiOperation(value = "Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 ", notes = "Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 ", response = void.class, authorizations = {
+    @io.swagger.annotations.ApiOperation(value = "Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 ", notes = "Fake endpoint for testing various parameters 假端點 偽のエンドポイント 가짜 엔드 포인트 ", response = Void.class, authorizations = {
         @io.swagger.annotations.Authorization(value = "http_basic_test")
     }, tags={ "fake", })
     @io.swagger.annotations.ApiResponses(value = { 
-        @io.swagger.annotations.ApiResponse(code = 400, message = "Invalid username supplied", response = void.class),
+        @io.swagger.annotations.ApiResponse(code = 400, message = "Invalid username supplied", response = Void.class),
```
Comment (jmatcuk) [inline samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/FakeApi.java]: Could you revert the formatting changes please.

Comment (amcginn) [reply samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/FakeApi.java]: This was done on purpose as "void.class" is not a class in Java, so I updated it to "Void.class."

Comment (wing328) [reply samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/FakeApi.java]: @amcginn I think that might introduce a minor issue:
```
--- a/samples/server/petstore/jaxrs-cxf/src/gen/java/io/swagger/api/PetApi.java
+++ b/samples/server/petstore/jaxrs-cxf/src/gen/java/io/swagger/api/PetApi.java
@@ -32,7 +32,7 @@ public interface PetApi  {
     @ApiOperation(value = "Add a new pet to the store", tags={ "pet",  })
     @ApiResponses(value = { 
         @ApiResponse(code = 405, message = "Invalid input") })
-    public void addPet(@Valid Pet body);
+    public Void addPet(@Valid Pet body);
```
`void` becomes `Void` in the method signature and the code won't compile.

Comment (amcginn) [reply samples/server/petstore/jaxrs/jersey2/src/gen/java/io/swagger/api/FakeApi.java]: I ran the jaxrs-cxf-petstore-server.sh while on master and this happened as well.

**Comment by amcginn** [comment]
Closing as this was addressed by another PR: #6373

**Pull request closed without merging**